diff --git a/dist/cjs/index.js b/dist/cjs/index.js
index 014828f9bfd46fe97fabd5238e98d05ca44b8bb4..3b6979e002ba58513e4eb8cb4f66b3da22a5d3f4 100644
--- a/dist/cjs/index.js
+++ b/dist/cjs/index.js
@@ -3043,8 +3043,8 @@ class DateRangeContext {
   render(data2, range, days) {
     var _a, _b, _c, _d;
     let result = null;
-    const startDayIndex = days[0].dayIndex;
-    const endDayIndex = days[days.length - 1].dayIndex;
+    const startDayIndex = days[0]?.dayIndex ?? -Infinity;
+    const endDayIndex = days[days.length - 1]?.dayIndex ?? Infinity;
     if (range.hasRepeat) {
       days.forEach((day) => {
         var _a2, _b2;
diff --git a/dist/es/index.js b/dist/es/index.js
index 606eda056616c69c10d295dd3edead8ef7d1b8f2..83203894767057b99c8a9099f215dc12a771aedb 100644
--- a/dist/es/index.js
+++ b/dist/es/index.js
@@ -3041,8 +3041,8 @@ class DateRangeContext {
   render(data2, range, days) {
     var _a, _b, _c, _d;
     let result = null;
-    const startDayIndex = days[0].dayIndex;
-    const endDayIndex = days[days.length - 1].dayIndex;
+    const startDayIndex = days[0]?.dayIndex ?? -Infinity;
+    const endDayIndex = days[days.length - 1]?.dayIndex ?? Infinity;
     if (range.hasRepeat) {
       days.forEach((day) => {
         var _a2, _b2;
diff --git a/dist/mjs/index.mjs b/dist/mjs/index.mjs
index e126fdd53406fee1408af3544a7707aeac023eb5..9e80e714ab9cf92002c861b66af40198304b3935 100644
--- a/dist/mjs/index.mjs
+++ b/dist/mjs/index.mjs
@@ -3041,8 +3041,8 @@ class DateRangeContext {
   render(data2, range, days) {
     var _a, _b, _c, _d;
     let result = null;
-    const startDayIndex = days[0].dayIndex;
-    const endDayIndex = days[days.length - 1].dayIndex;
+    const startDayIndex = days[0]?.dayIndex ?? -Infinity;
+    const endDayIndex = days[days.length - 1]?.dayIndex ?? Infinity;
     if (range.hasRepeat) {
       days.forEach((day) => {
         var _a2, _b2;
